### PR TITLE
chore: changes from dev into main

### PR DESCRIPTION
**Pull request by wainola**
## Description

Bringing changes from dev into main

## Related Issue Or Context

Closes: #<issue>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in sygma-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Comment by mpetrunic** [comment]
Please do not merge, once approved, I will create merge commit on main branch^^

**Review by mpetrunic — APPROVED** [review]

File: packages/widget/src/components/address-input/address-input.ts
Comment (Lykhoyda) [inline packages/widget/src/components/address-input/address-input.ts:79]: Are you sure that it shouldn't be a textAreaElement?

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:194]: probably should be removed as we init the config inside retryInitSdk

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:173]: I think it was wrongly overwritten, the entire story https://github.com/sygmaprotocol/sygma-widget/pull/176/files which was merged to master is reverted

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:45]: it's the correct version from https://github.com/sygmaprotocol/sygma-widget/pull/176/files

File: packages/widget/src/components/address-input/address-input.ts
Comment (wainola) [reply packages/widget/src/components/address-input/address-input.ts]: solved

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts:173]: I restore the changes, could you please let me know if there is anything missing? I think not but better to have two pair of eyes here

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts]: solved

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts]: it's a duplication, we already have this code above

**Comment by wainola** [comment]
> please double check the code changes, I spotted that the bug fix from https://github.com/sygmaprotocol/sygma-widget/pull/176/files was almost everywhere reverted to the state of the dev branch. I'm not sure about other stories, but it worth to check again

I restore last missing changes that were badly overwritten

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:223]: The comment should stay for consistency ;)

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts]: removed

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (Lykhoyda) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:296]: looks like duplication as well

Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts]: resolved

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts:193]: got it and solved

**Review by saadjhk — DISMISSED** [review]

**Comment by Lykhoyda** [comment]
@wainola deployed version  https://d633b65c.sygma-widget.pages.dev/ has infinite loader, something is wrong with initSdk state probably 

Screenshot 2024-04-24 at 18 08 09

File: packages/widget/src/controllers/transfers/fungible-token-transfer.ts
Comment (saadjhk) [inline packages/widget/src/controllers/transfers/fungible-token-transfer.ts:193]: `await this.retryInitSdk();` why is there an init with hardcoded `chainId`

Comment (wainola) [reply packages/widget/src/controllers/transfers/fungible-token-transfer.ts:194]: solved

**Comment by wainola** [comment]
> @wainola deployed version https://d633b65c.sygma-widget.pages.dev/ has infinite loader, something is wrong with initSdk state probably
> 
> Screenshot 2024-04-24 at 18 08 09

fixed @Lykhoyda

**Comment by Lykhoyda** [comment]
@wainola have you been able to finish the transfer? For me, it stuck on the: 
cannot estimate gas; transaction may fail or may require manual gas limit

Screenshot 2024-04-25 at 16 50 28

**Review by Lykhoyda — APPROVED** [review]

**Comment by wainola** [comment]
> @wainola have you been able to finish the transfer? For me, it stuck on the: cannot estimate gas; transaction may fail or may require manual gas limit
> 
> Screenshot 2024-04-25 at 16 50 28

@Lykhoyda solved, could you re-test on your side?

**Review by Lykhoyda — APPROVED** [review]

**Pull request merged**
